File: C++/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <iostream>
#include <vector>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    vector<char> v;
    
    for(char c : s){
        if (c == '('){
            v.push_back('(');
        }
        else{
            if (v.size() == 0){
                answer = false;
                break;
            }
            else{
                v.pop_back();
            }
        }
    }
    
    if (answer && v.size() != 0){
        answer = false;
    }
    

    return answer;
}

File: Python3/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    tmp = []
    
    for i in s:
        if i == "(":
            tmp.append("(")
        else:
            if not tmp:
                answer = False
                break
            else:
                tmp.pop(-1)
                
    if answer == True and tmp:
        answer = False

    return answer